### PR TITLE
Add HNPWA- Ember.js implementation (hackernews.io)

### PR DESCRIPTION
**Pull request by ivanvanderbyl**
This PR adds an Ember.js implementation, as mentioned in #44 

**Results:**

- First Interactive: Emerging markets: [4.727s](https://www.webpagetest.org/result/171007_GJ_52584ab4b5e5cf2978d2a20cb28a74e8/)
- First Interactive:  Faster 3G Moto G4: [4.254s](https://www.webpagetest.org/result/171007_3Q_d90b70ffa917bf4ff7300d71575c4262/)
Lighthouse score: [100/100](https://googlechrome.github.io/lighthouse/viewer/?gist=c358a609038f463cb06741b2847f5315)

Demo: [hackernews.io](https://hackernews.io)

### CODE REVIEW

**Comment by addyosmani** [comment]
Thanks for putting this together, @ivanvanderbyl!

Kicking off some WPT runs in the hope they complete later:

Faster 3G: https://www.webpagetest.org/result/170606_6X_1022022a29ea38db18854d151269bbc5/
EM: https://www.webpagetest.org/result/170606_0N_d5c53eaed714f730fd3706423405ea32/

**Comment by ivanvanderbyl** [comment]
Interesting, that 3G  test spends a lot of time processing something. I'll investigate why this is, there might be a faster way I can render this list.

**Comment by addyosmani** [comment]
I kicked off another run on faster 3G as it was a little surprising the EM test got interactivity data back but that one didn't. https://www.webpagetest.org/result/170606_PF_b72334ca1965e803f11f2fc3371c216a/

File: site/_apps/ember-hn.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+module-bundling: 'Broccoli'
+service-worker: 'Application Shell + data caching'
+perfomance-patterns: 'Asynchronous Partial Outputs'
+server-side-rendering: 'No'
+api: 'Node-hnapi (unofficial)'
+hosting: 'Cloudfront'
+authors:
+    - name: 'Ivan'
+lighthouse: '91/100'
+interactive-em: '3.1s'
```
Comment (addyosmani) [inline site/_apps/ember-hn.txt]: EM first-interactive according to https://www.webpagetest.org/lighthouse.php?test=170607_FT_95b840f28ed8a72b031a8a06eb24ec68&run=3 (click through to the Lighthouse report on https://www.webpagetest.org/result/170607_FT_95b840f28ed8a72b031a8a06eb24ec68/) is 7.3s

File: site/_apps/ember-hn.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+interactive-faster-3g: '2.0s'
+lighthouse-link:
+wpt-em-link: https://www.webpagetest.org/result/170511_AW_b2e2360d4207831ece22ab4210ab9444/
+wpt-faster-3g-link: https://www.webpagetest.org/result/170511_7Q_15882b2c52ed51c614ec707736e52940/
```
Comment (addyosmani) [inline site/_apps/ember-hn.txt]: Fresh run on faster 3G: 
https://www.webpagetest.org/result/170607_SC_e59cf850f6e95cf03643cc66cfe50bdb/ 7,430ms (7.4s) for first interactive. 

If we take a look at the filmstrip what we see is that there isn't any content rendered up until around the 7s or so point: https://www.webpagetest.org/video/compare.php?tests=170607_SC_e59cf850f6e95cf03643cc66cfe50bdb-r%3A1-c%3A0&thumbSize=200&ival=500&end=full

**Comment by addyosmani** [comment]
EM first-interactive according to https://www.webpagetest.org/lighthouse.php?test=170607_FT_95b840f28ed8a72b031a8a06eb24ec68&run=3 (click through to the Lighthouse report on https://www.webpagetest.org/result/170607_FT_95b840f28ed8a72b031a8a06eb24ec68/) is 7.3s

Fresh run on faster 3G:
https://www.webpagetest.org/result/170607_SC_e59cf850f6e95cf03643cc66cfe50bdb/ 7,430ms (7.4s) for first interactive. I'm going to chalk this being slower than the EM number up to latencies and we should re-run.

If we take a look at the filmstrip what we see is that there isn't any content rendered up until around the 7s or so point: https://www.webpagetest.org/video/compare.php?tests=170607_SC_e59cf850f6e95cf03643cc66cfe50bdb-r%3A1-c%3A0&thumbSize=200&ival=500&end=full

-------

I think it's worth checking how much of this is server-latency related and whether there are further optimizations that could be applied.

**Comment by addyosmani** [comment]
I just saw that @ivanvanderbyl put together https://github.com/ivanvanderbyl/ember-hackernews-pwa/issues/1. Doesn't look like it's _too_ far off now 😄 Do let us know if you run into any further issues with WebPageTest/Lighthouse at all.

**Comment by addyosmani** [comment]
Periodically checking in on this app: looks like there are two items left to go upstream. Looking forward to seeing it when complete. Exciting.

**Comment by ivanvanderbyl** [comment]
Hey guys, sorry been really crazy flat out on this end with real work, I'll try to get this polished on one weekend soon.

**Comment by addyosmani** [comment]
Thanks, @ivanvanderbyl! 🎉

**Comment by ivanvanderbyl** [comment]
Ping @addyosmani  updated with newer results

**Comment by ivanvanderbyl** [comment]
Ping @addyosmani updated again with even better results after spending a few days performance tuning, got time to interactive to sub 5s, best result was 3.9s, but lighthouse score was below 90, so I've included a report at 100/100 and 4.2s! 💯

**Comment by addyosmani** [comment]
LGTM. Great work!

**Pull request merged**